infrastructure/smart-contracts/contracts/upgrades/pool-factory/PoolFactory2.sol

```diff
@@ -616,6 +632,20 @@ contract PoolFactory2 is Initializable, AccessControlEnumerableUpgradeable {
             removeUserFromPool(msg.sender, pool);
         }
 
+        // Update total stake
+        if(!totalStakeCalculated[msg.sender]) {
+            uint256 totalPools = interactedPoolsOfUser[msg.sender].length;
+            // We check the total pools interacted with by the user to avoid running out of gas
+            // This ensures we do not run this one time calculation until we are sure that it will not run out of gas
+            // This ensures a user can always unstake their esXai
+            if(totalPools < 150) {
```

Comment:
Did we arbitrarily pick `150` here? If so perhaps we should make this a state variable with a setter function in case it needs to be smaller. That way we can update it with a simple trx instead of having to redeploy the whole contract.

This seems like an extreme edge case though so the juice may not be worth the squeeze here. I'll leave it up to you 👍

Reply:
Yes, it was arbitrary, I think we could get away with 250+, but might be something we should test in the test cases just to be sure? Like you say, may or may not be worth the effort.

As long as we know that a tx will not fail at that number, we should never need to change it.